scripts/tick_my_feedstocks.py

```diff
@@ -371,29 +423,48 @@ def pypi_org_sha(package_name, version):
     return None, None
 
 
-def pypi_sha(pypi_package, pypi_version):
+def sha(package, version, package_type='pypi', package_url=None,
+        prior_version=None):
     """
-    :param str pypi_package: The name of the package in PyPI
-    :param str pypi_version: The version to be retrieved from PyPI.
+    :param str package: The name of the package in PyPI
+    :param str version: The version to be retrieved from PyPI.
     :return: `str|None` -- SHA256 for a source bundle, None if can't be found
     """
-    bundle_type, sha = pypi_legacy_json_sha(pypi_package, pypi_version)
-    if bundle_type is not None and sha is not None:
-        return bundle_type, sha
-    return pypi_org_sha(pypi_package, pypi_version)
+    if package_type == 'github':
+        package_url = package_url.replace(prior_version, version)
+        return hash_url(package_url)
+    else:
+        bundle_type, sha = pypi_legacy_json_sha(package, version)
+        if bundle_type is not None and sha is not None:
+            return bundle_type, sha
+        return pypi_org_sha(package, version)
 
 
-def pypi_version_str(package_name):
+def version_str(package_name, package_type='pypi', package_owner=None):
     """
     Retrive the latest version of a package in PyPI
     :param str package_name: The name of the package
+    :param str package_type: The type of package ('pypi' or 'github')
     :return: `str|bool` -- Version string, False if unsuccessful
     """
-    r = requests.get('https://pypi.python.org/pypi/{}/json'.format(
-        package_name))
-    if not r.ok:
-        return False
-    return r.json()['info']['version'].strip()
+    if package_type == 'pypi':
+        r = requests.get('https://pypi.python.org/pypi/{}/json'.format(
+            package_name))
+        if not r.ok:
+            return False
+        return r.json()['info']['version'].strip()
+    elif package_type == 'github':
+        # get all the tags
+        refs = requests.get('https://api.github.com/repos/{owner}/'
+                         '{repo}/git/refs/tags'.format(owner=package_owner,
+                                                       repo=package_name))
```

Comment:
Is this package name taken from the conda package name or from the github url ?

Reply:
From the meta.yaml, source, fn, in the same way as the pypi implementation.

Reply:
Ah okay. That's a separate issue then.

Reply:
I could parse the URL if that is preferred.

Reply:
There are issues with the PyPI implementation as well because the conda package name and PyPI package may not be the same. This happens rarely.
I expect this to happen frequently with non python github packages.

Reply:
That's fair.

Reply:
I guess we should really split how this is handled with two package names, one listed on the feedstock and one for the remote source of the code.

Reply:
Although I would also maybe push for putting a definitive copy of this information someplace (maybe in about?)

Reply:
There was some talk about this at https://github.com/conda-forge/feedstocks/issues/12